### PR TITLE
Replace jQuery scrollspy

### PR DESCRIPTION
**Pull request by imeugenia**

### CODE REVIEW

**Comment by skibish** [comment]
I see, that it's currently in progress (label), so I will not look at it right now :)

But I really like the intention to remove jQuery :+1: :100:

File: themes/falconf/src/js/colorChangeOnScroll.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const isScrolledToBottom = (element, scrolledFromTop) => {
+    const bottom = element.offsetTop + element.offsetHeight
+    return scrolledFromTop === bottom - window.innerHeight 
+}
+
+const getClassName = id => `active-${id}`
+
+const resolveClass = (currentElement, index, scrolledFromTop) => {
+    const isFirst = currentElement.offsetTop === 0 ? true : false
```
Comment (skibish) [inline themes/falconf/src/js/colorChangeOnScroll.js]: You can simplify this to

```js

const isFirst = currentElement.offsetTop === 0

```

File: themes/falconf/src/js/colorChangeOnScroll.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    if (isFirst) {
```
Comment (skibish) [inline themes/falconf/src/js/colorChangeOnScroll.js]:
```js

if (isFirst && isOnTopSection(scrolledFromTop)) {
  return '';
}

```

File: themes/falconf/src/js/colorChangeOnScroll.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        if (isOnTopSection(scrolledFromTop)) {
+            return ''
+        }
+    }
+
+    const isLast = index === SECTIONS.length - 1 ? true : false
```
Comment (skibish) [inline themes/falconf/src/js/colorChangeOnScroll.js]:
```js

const isLast = index === SECTIONS.length - 1

```

File: themes/falconf/src/js/colorChangeOnScroll.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    if (isLast) {
```
Comment (skibish) [inline themes/falconf/src/js/colorChangeOnScroll.js]:
```js

if (isLast && isScrolledToBottom(currentElement, scrolledFromTop)) {
  return getClassName(currentElement.id);
}

const top = getColorChangeBorder(currentElement.offsetTop, currentElement.offsetHeight);
const nextElement = SECTIONS[index + 1];
const bottom = getColorChangeBorder(nextElement.offsetTop, nextElement.offsetHeight);

if (scrolledFromTop >= top && scrolledFromTop < bottom) {
  return getClassName(nextElement.id);
}

// and you want to return here something
// return ''; // ?

```

File: themes/falconf/src/js/colorChangeOnScroll.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    const scrolledFromTop = window.scrollY
+    SECTIONS.forEach((currentElement, index) => {
+        const newClass = resolveClass(currentElement, index, scrolledFromTop)
+        if(newClass !== undefined) {
+            body.className = newClass
+        }
+    })
+}
+
+window.addEventListener('scroll', handleScroll)
```
Comment (skibish) [inline themes/falconf/src/js/colorChangeOnScroll.js]: I would suggest to make this functionality as a module. You will have something like this in the end:

```js
// js/colorChangeOnScroll.js

export default function() {
  // logic that bootstrap and return handleScroll
}

```
And in main file you will trigger it
```js
// index.js

import colorChangeOnScroll from './colorChangeOnScroll.js');

// ...

 $(function (){
  window.addEventListener('scroll', colorChangeOnScroll)
  // ...
 })	 })

```

File: themes/falconf/src/js/colorChangeOnScroll.js (hunk 4 above)
Comment (imeugenia) [reply themes/falconf/src/js/colorChangeOnScroll.js]: Nope

**Pull request merged**